smdebug/tensorflow/keras.py

```diff
@@ -1239,12 +1239,19 @@ def run(*args, **kwargs):
                         and hasattr(vars[0], "numpy")
                     )
                 )
-                or (not ((isinstance(grads[0], tf.Tensor)) and hasattr(grads[0], "numpy")))
             ):
                 return grads
+            # get embedding layers indexes
+            embedding_grad_indexes = set()
+            for idx, grad in enumerate(grads):
+                if not (isinstance(grad, tf.Tensor) and hasattr(grad, "numpy")):
+                    if isinstance(grad, tf.IndexedSlices):
+                        embedding_grad_indexes.add(idx)
+                    else:
+                        return grads
```

Comment:
This logic and using of `set` is hard to read and understand. It can be simplified: 
```
# if we have unknown object type then do nothing
if any(map(lambda grad: not (is_valid_tensor(grad) ||  is_IndexedSlices(grad), grads)):
    return grads
```